project_task_parent_due_auto/models/project_task.py

```diff
@@ -0,0 +1,36 @@
+from datetime import date
+
+from odoo import api, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    @api.model
+    def create(self, values):
+        tasks = super().create(values)
+        if "date_deadline" in values:
+            tasks.update_parent_task_dates()
+        return tasks
```

Comment:
You are decreasing the performance not doing it multi-batch and mixing some concepts:

```suggestion
    @api.model_create_multi
    def create(self, vals_list):
        tasks = super().create(vals_list)
        for vals, task in zip(vals_list, tasks):
            if "date_deadline" in vals:
                task.update_parent_task_dates()
        return tasks
```